source/log.hpp

```diff
@@ -35,30 +44,32 @@ namespace reshade::log
 		inline message &operator<<(const T &value)
 		{
 			stream << value;
+			linestream << value;
+			
 			return *this;
 		}
+		inline message &operator<<(const char *message)
+		{
+			stream << message;
+			linestream << message;
 
+			return *this;
+		}
 		template <>
 		inline message &operator<<(const std::wstring &message)
 		{
-			static_assert(sizeof(std::wstring::value_type) == sizeof(uint16_t), "expected 'std::wstring' to use UTF-16 encoding");
-			std::string utf8_message;
-			utf8_message.reserve(message.size());
-			utf8::unchecked::utf16to8(message.begin(), message.end(), std::back_inserter(utf8_message));
-			return operator<<(utf8_message);
-		}
+			std::string utf8message;
+			utf8::unchecked::utf16to8(message.begin(), message.end(), std::back_inserter(utf8message));
+			return operator<<(utf8message);
```

Comment:
What's the reasoning behind removing the `reserve` call? `std::back_inserter` uses `push_back` which greatly benefits from a pre-allocated storage space.

Reply:
oops - this was just a mistake --all three of these lines (44-46) shouldn't have been deleted. I was using an older log.hpp file and didn't realize you had already fixed these.